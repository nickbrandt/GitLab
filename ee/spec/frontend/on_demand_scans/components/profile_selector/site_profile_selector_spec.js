import { mount, shallowMount } from '@vue/test-utils';
import { merge } from 'lodash';
import ProfileSelector from 'ee/on_demand_scans/components/profile_selector/profile_selector.vue';
import OnDemandScansSiteProfileSelector from 'ee/on_demand_scans/components/profile_selector/site_profile_selector.vue';
import SiteProfileSummary from 'ee/on_demand_scans/components/profile_selector/site_profile_summary.vue';
import { siteProfiles } from '../../mocks/mock_data';

const TEST_LIBRARY_PATH = '/test/site/profiles/library/path';
const TEST_NEW_PATH = '/test/new/site/profile/path';
const TEST_ATTRS = {
  'data-foo': 'bar',
};
const profiles = siteProfiles.map((x) => {
  return {
    ...x,
    dropdownLabel: `${x.profileName}: ${x.targetUrl}`,
  };
});

describe('OnDemandScansSiteProfileSelector', () => {
  let wrapper;

  const wrapperFactory = (mountFn = shallowMount) => (options = {}) => {
    wrapper = mountFn(
      OnDemandScansSiteProfileSelector,
      merge(
        {
          propsData: {
            profiles: [],
          },
          attrs: TEST_ATTRS,
          provide: {
            siteProfilesLibraryPath: TEST_LIBRARY_PATH,
            newSiteProfilePath: TEST_NEW_PATH,
          },
          slots: {
            summary: `<div>${profiles[0].profileName}'s summary</div>`,
          },
        },
        options,
      ),
    );
  };
  const createComponent = wrapperFactory();
  const createFullComponent = wrapperFactory(mount);

  const findProfileSelector = () => wrapper.find(ProfileSelector);

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders properly with profiles', () => {
    createFullComponent({
      propsData: { profiles, value: profiles[0].id },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders properly without profiles', () => {
    createFullComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  describe('profile summary', () => {
    it('is rendered when a profile is selected', () => {
      const selectedProfile = profiles[0];

      createComponent({
        propsData: { profiles, value: selectedProfile.id, selectedProfile },
      });

      expect(wrapper.findComponent(SiteProfileSummary).exists()).toBe(true);
    });

    it('is not rendered when no profile is selected', () => {
      createComponent({
        propsData: { profiles, selectedProfile: null },
      });

      expect(wrapper.findComponent(SiteProfileSummary).exists()).toBe(false);
    });
  });

  it('sets listeners on profile selector component', () => {
    const inputHandler = jest.fn();
    createComponent({
      listeners: {
        input: inputHandler,
      },
    });
    findProfileSelector().vm.$emit('input');

    expect(inputHandler).toHaveBeenCalled();
  });

  describe('with profiles', () => {
    it('renders profile selector', () => {
      createComponent({
        propsData: { profiles },
      });
      const sel = findProfileSelector();

      expect(sel.props()).toEqual({
        libraryPath: TEST_LIBRARY_PATH,
        newProfilePath: TEST_NEW_PATH,
        profiles,
        value: null,
      });
      expect(sel.attributes()).toMatchObject(TEST_ATTRS);
    });
  });
});
