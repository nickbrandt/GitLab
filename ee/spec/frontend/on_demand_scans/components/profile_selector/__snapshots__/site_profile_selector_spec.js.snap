// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnDemandScansSiteProfileSelector renders properly with profiles 1`] = `
<div
  class="gl-card"
  data-foo="bar"
  value="gid://gitlab/DastSiteProfile/1"
>
  <div
    class="gl-card-header"
  >
    <div
      class="row"
    >
      <div
        class="col-7 gl-display-flex gl-align-items-center"
      >
        <h3
          class="gl-font-lg gl-my-0"
        >
          Site profile
        </h3>
      </div>
    </div>
  </div>
   
  <div
    class="gl-card-body"
  >
     
    <fieldset
      aria-describedby=""
      class="form-group gl-form-group"
    >
      <legend
        class="bv-no-focus-ring col-form-label pt-0 col-form-label"
        tabindex="-1"
      >
        Use existing site profile
      </legend>
      <div
        class="bv-no-focus-ring"
        role="group"
        tabindex="-1"
      >
         
        <div
          class="dropdown b-dropdown gl-new-dropdown mw-460 btn-group"
          data-testid="profiles-dropdown"
        >
          <!---->
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="btn dropdown-toggle btn-default btn-md gl-button gl-dropdown-toggle"
            type="button"
          >
            <!---->
             
            <!---->
             
            <span
              class="gl-new-dropdown-button-text"
            >
              Site profile #1: https://foo.com
            </span>
             
            <svg
              aria-hidden="true"
              class="gl-button-icon dropdown-chevron gl-icon s16"
              data-testid="chevron-down-icon"
              role="img"
            >
              <use
                href="#chevron-down"
              />
            </svg>
          </button>
          <ul
            class="dropdown-menu"
            role="menu"
            tabindex="-1"
          >
            <div
              class="gl-new-dropdown-inner"
            >
              <div
                class="gl-new-dropdown-header gl-border-b-0!"
              >
                <!---->
                 
                <div
                  class="gl-search-box-by-type"
                >
                  <svg
                    aria-hidden="true"
                    class="gl-search-box-by-type-search-icon gl-icon s16"
                    data-testid="search-icon"
                    role="img"
                  >
                    <use
                      href="#search"
                    />
                  </svg>
                   
                  <input
                    aria-label="Search"
                    class="gl-form-input gl-search-box-by-type-input form-control"
                    placeholder="Search"
                    type="text"
                  />
                   
                  <div
                    class="gl-search-box-by-type-right-icons"
                  >
                    <!---->
                     
                    <!---->
                  </div>
                </div>
              </div>
               
              <div
                class="gl-new-dropdown-contents"
              >
                 
                <li
                  class="gl-new-dropdown-item"
                  role="presentation"
                >
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="gl-icon s16 gl-new-dropdown-item-check-icon gl-mt-3 gl-align-self-start"
                      data-testid="dropdown-item-checkbox"
                      role="img"
                    >
                      <use
                        href="#mobile-issue-close"
                      />
                    </svg>
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="gl-new-dropdown-item-text-wrapper"
                    >
                      <p
                        class="gl-new-dropdown-item-text-primary"
                      >
                        
        Site profile #1
      
                      </p>
                       
                      <!---->
                    </div>
                     
                    <!---->
                  </button>
                </li>
                <li
                  class="gl-new-dropdown-item"
                  role="presentation"
                >
                  <button
                    class="dropdown-item"
                    role="menuitem"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="gl-icon s16 gl-new-dropdown-item-check-icon gl-visibility-hidden gl-mt-3 gl-align-self-start"
                      data-testid="dropdown-item-checkbox"
                      role="img"
                    >
                      <use
                        href="#mobile-issue-close"
                      />
                    </svg>
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="gl-new-dropdown-item-text-wrapper"
                    >
                      <p
                        class="gl-new-dropdown-item-text-primary"
                      >
                        
        Site profile #2
      
                      </p>
                       
                      <!---->
                    </div>
                     
                    <!---->
                  </button>
                </li>
                 
                <div
                  class="gl-p-3 gl-text-center"
                  style="display: none;"
                >
                  
        No matching results...
      
                </div>
              </div>
               
              <div
                class="gl-new-dropdown-footer"
              >
                <li
                  class="gl-new-dropdown-item"
                  role="presentation"
                >
                  <a
                    class="dropdown-item"
                    data-testid="create-profile-option"
                    href="/test/new/site/profile/path"
                    role="menuitem"
                    target="_self"
                  >
                    <!---->
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="gl-new-dropdown-item-text-wrapper"
                    >
                      <p
                        class="gl-new-dropdown-item-text-primary"
                      >
                        Create new site profile
                      </p>
                       
                      <!---->
                    </div>
                     
                    <!---->
                  </a>
                </li>
                 
                <li
                  class="gl-new-dropdown-item"
                  role="presentation"
                >
                  <a
                    class="dropdown-item"
                    data-testid="manage-profiles-option"
                    href="/test/site/profiles/library/path"
                    role="menuitem"
                    target="_self"
                  >
                    <!---->
                     
                    <!---->
                     
                    <!---->
                     
                    <div
                      class="gl-new-dropdown-item-text-wrapper"
                    >
                      <p
                        class="gl-new-dropdown-item-text-primary"
                      >
                        Manage site profiles
                      </p>
                       
                      <!---->
                    </div>
                     
                    <!---->
                  </a>
                </li>
              </div>
            </div>
          </ul>
        </div>
         
        <div
          class="gl-mt-6 gl-pt-6 gl-border-t-solid gl-border-gray-100 gl-border-t-1"
          data-testid="selected-profile-summary"
        >
          <a
            aria-label="Edit profile"
            class="btn gl-absolute gl-right-7 gl-z-index-1 btn-default btn-md gl-button btn-icon"
            href="/site_profiles/edit/1"
            title="Edit profile"
          >
            <!---->
             
            <svg
              aria-hidden="true"
              class="gl-button-icon gl-icon s16"
              data-testid="pencil-icon"
              role="img"
            >
              <use
                href="#pencil"
              />
            </svg>
              
            <!---->
          </a>
           
        </div>
        <!---->
        <!---->
        <!---->
      </div>
    </fieldset>
  </div>
   
  <!---->
</div>
`;

exports[`OnDemandScansSiteProfileSelector renders properly without profiles 1`] = `
<div
  class="gl-card"
  data-foo="bar"
>
  <div
    class="gl-card-header"
  >
    <div
      class="row"
    >
      <div
        class="col-7 gl-display-flex gl-align-items-center"
      >
        <h3
          class="gl-font-lg gl-my-0"
        >
          Site profile
        </h3>
      </div>
    </div>
  </div>
   
  <div
    class="gl-card-body"
  >
     
    <p
      class="gl-text-gray-700"
    >
      No profile yet. In order to create a new scan, you need to have at least one completed site profile.
    </p>
     
    <a
      class="btn btn-confirm btn-md gl-button btn-confirm-secondary"
      data-testid="create-profile-link"
      href="/test/new/site/profile/path"
    >
      <!---->
       
      <!---->
        
      <span
        class="gl-button-text"
      >
        Create new site profile
      </span>
    </a>
  </div>
   
  <!---->
</div>
`;
